src/libraries/System.Private.CoreLib/src/System/IndexOfAnyAsciiSearcher.cs

```diff
@@ -0,0 +1,333 @@
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int ComputeFirstIndexOverlapped<TNegator>(ref short searchSpace, ref short current0, ref short current1, Vector128<byte> result)
+            where TNegator : struct, INegator
+        {
+            uint mask = TNegator.ExtractMask(result);
+            int offsetInVector = BitOperations.TrailingZeroCount(mask);
+            if (offsetInVector >= Vector128<short>.Count)
+            {
+                // We matched within the second vector
+                current0 = ref current1;
+                offsetInVector -= Vector128<short>.Count;
+            }
+            return offsetInVector + (int)(Unsafe.ByteOffset(ref searchSpace, ref current0) / sizeof(short));
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int ComputeLastIndex<TNegator>(ref short searchSpace, ref short current, Vector128<byte> result)
+            where TNegator : struct, INegator
+        {
+            uint mask = TNegator.ExtractMask(result) & 0xFFFF;
+            int offsetInVector = 31 - BitOperations.LeadingZeroCount(mask);
+            return offsetInVector + (int)(Unsafe.ByteOffset(ref searchSpace, ref current) / sizeof(char));
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int ComputeLastIndexOverlapped<TNegator>(ref short searchSpace, ref short secondVector, Vector128<byte> result)
+            where TNegator : struct, INegator
+        {
+            uint mask = TNegator.ExtractMask(result) & 0xFFFF;
+            int offsetInVector = 31 - BitOperations.LeadingZeroCount(mask);
+            if (offsetInVector < Vector128<short>.Count)
+            {
+                return offsetInVector;
+            }
+
+            // We matched within the second vector
+            return offsetInVector - Vector128<short>.Count + (int)(Unsafe.ByteOffset(ref searchSpace, ref secondVector) / sizeof(char));
+        }
+
+        private interface INegator
+        {
+            static abstract bool NegateIfNeeded(bool result);
+            static abstract Vector128<byte> NegateIfNeeded(Vector128<byte> result);
+            static abstract uint ExtractMask(Vector128<byte> result);
+        }
+
+        private struct DontNegate : INegator
+        {
+            public static bool NegateIfNeeded(bool result) => result;
+            public static Vector128<byte> NegateIfNeeded(Vector128<byte> result) => result;
+            public static uint ExtractMask(Vector128<byte> result) => ~Vector128.Equals(result, Vector128<byte>.Zero).ExtractMostSignificantBits();
+        }
+
+        private struct Negate : INegator
+        {
+            public static bool NegateIfNeeded(bool result) => !result;
+            public static Vector128<byte> NegateIfNeeded(Vector128<byte> result) => Vector128.Equals(result, Vector128<byte>.Zero);
```

Comment:
```suggestion
            public static Vector128<byte> NegateIfNeeded(Vector128<byte> result) => ~result;
```

Reply:
These aren't equivalent. I explicitly want to check whether any value was 0.
The name may not be the best though, suggestions are welcome :)

Reply:
Got you, but I don't have any real idea for a better name. But a comment explaining to check for 0 here is the intention would clarify this.

For names maybe `CompareToZeroIfNeeded`? This doesn't fit to `INegator` anyway. I think as internal implementation detail the comment is the best option -- except someone else comes up with a proper name.